sw/device/tests/sim_dv/rv_dm_ndm_reset_req.c

```diff
@@ -0,0 +1,262 @@
+static dif_keymgr_t keymgr;
+
+enum {
+  OTP_CTRL,
+  PINMUX,
+  SRAM_CTRL_RET,
+  ADC_CTRL,
+  SYSRST_CTRL,
+  KEYMGR,
+  FLASH_CTRL,
+};
+
+static test_register_t kReg[] = {
+    [OTP_CTRL] =
+        {
+            .name = "OTP_CTRL",
+            .dif_handle = &otp_ctrl,
+            .write_val = 0x06092022,
+            .exp_read_val = 0x06092022,
+        },
+    [PINMUX] =
+        {
+            .name = "PINMUX",
+            .write_val = 0x44,
+            .exp_read_val = 0x44,
+            .dif_handle = &pinmux,
+        },
+    [SRAM_CTRL_RET] =
+        {
+            .name = "SRAM_CTRL_RET",
+            .write_val = 0xddAA55bb,
+            .exp_read_val = 0xddAA55bb,
+        },
+    [ADC_CTRL] =
+        {
+            .name = "ADC_CTRL",
+            .write_val = 0x37,
+            .exp_read_val = 0x9b,
+            .dif_handle = &adc_ctrl,
+        },
+    [SYSRST_CTRL] =
+        {
+            .name = "SYSRST_CTRL",
+            .write_val = 0x567,
+            .exp_read_val = 0x7d0,
+            .dif_handle = &sysrst_ctrl,
+        },
+    [KEYMGR] =
+        {
+            .name = "KEYMGR",
+            .write_val = 0x1600ABBA,
+            .exp_read_val = 0x0,
+            .dif_handle = &keymgr,
+        },
+    [FLASH_CTRL] =
+        {
+            .name = "FLASH_CTRL",
+            .dif_handle = &flash_ctrl,
+            .write_val = 0x3927,
+            .exp_read_val = 0x0,
+        },
+};
+
+static void write_test_reg() {
+  mmio_region_write32(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR),
+      OTP_CTRL_DIRECT_ACCESS_WDATA_0_REG_OFFSET, kReg[OTP_CTRL].write_val);
+  mmio_region_write32(
+      mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR),
+      FLASH_CTRL_SCRATCH_REG_OFFSET, kReg[FLASH_CTRL].write_val);
+  mmio_region_write32(
+      mmio_region_from_addr(TOP_EARLGREY_SRAM_CTRL_RET_AON_RAM_BASE_ADDR),
+      8 * sizeof(uint32_t), kReg[SRAM_CTRL_RET].write_val);
+  mmio_region_write32(
+      mmio_region_from_addr(TOP_EARLGREY_ADC_CTRL_AON_BASE_ADDR),
+      ADC_CTRL_ADC_SAMPLE_CTL_REG_OFFSET, kReg[ADC_CTRL].write_val);
+  mmio_region_write32(
+      mmio_region_from_addr(TOP_EARLGREY_SYSRST_CTRL_AON_BASE_ADDR),
+      SYSRST_CTRL_EC_RST_CTL_REG_OFFSET, kReg[SYSRST_CTRL].write_val);
+  mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR),
+                      PINMUX_WKUP_DETECTOR_CNT_TH_1_REG_OFFSET,
+                      kReg[PINMUX].write_val);
+  mmio_region_write32(mmio_region_from_addr(TOP_EARLGREY_KEYMGR_BASE_ADDR),
+                      KEYMGR_MAX_OWNER_KEY_VER_SHADOWED_REG_OFFSET,
+                      kReg[KEYMGR].write_val);
+}
+
+static uint32_t read_test_reg(uint32_t device_idx, bool is_initial) {
+  mmio_region_t addr;
+  ptrdiff_t offset;
+  switch (device_idx) {
+    case OTP_CTRL:
+      addr = mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR);
+      offset = OTP_CTRL_DIRECT_ACCESS_WDATA_0_REG_OFFSET;
```

Comment:
(just a question / suggestion) Would it make sense to throw these in the `test_register_t` table above? It seems like that would keep the values and offsets together, improving readability.

But it's also fine as-is.